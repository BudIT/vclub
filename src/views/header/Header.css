.header {
  margin: 0;
  padding: 0;
  display: flex;
  box-shadow: 0 0 10px black;
  z-index: 10;
}
