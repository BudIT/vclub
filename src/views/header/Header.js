import React, { PropTypes } from 'react';
import compose from 'recompose/compose';
import { connect } from 'react-redux';

// subviews
import HeaderRight from './HeaderRight/HeaderRight';
import HeaderLeft from './HeaderLeft/HeaderLeft';

import style from './Header.css';

const enhance = compose(
  connect(state => ({
    numberOfMembers: state.members.length,
    currentRoomName: state.rooms.currentRoom,
    user: state.auth.user,
  })),
);

function Header(props) {
  const roomsNames = ['SHARING', 'VIDEO', 'CHAT', 'WHITEBOARD'];

  const {
    dispatch,
    numberOfMembers,
    currentRoomName,
    user,
  } = props;


  return (
    <div className={style.header}>
      <HeaderLeft
        currentRoomName={currentRoomName}
        roomsNames={roomsNames}
        user={user}
        dispatch={dispatch}
      />
      <HeaderRight
        numberOfMembers={numberOfMembers}
        dispatch={dispatch}
      />
    </div>
  );
}

Header.propTypes = {
  dispatch: PropTypes.func.isRequired,
  numberOfMembers: PropTypes.number.isRequired,
  currentRoomName: PropTypes.string.isRequired,
  user: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    master: PropTypes.bool.isRequired,
  }).isRequired,
};

export default enhance(Header);
