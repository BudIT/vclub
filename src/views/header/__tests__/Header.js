/* eslint import/no-extraneous-dependencies: ["error", {"devDependencies": true}] */
/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { HeaderComponent } from '../Header';


const currentRoomName = 'VIDEO';
const numberOfMembers = 5;

const user = {
  id: '1',
  name: 'User',
  master: true,
};

test('<Header /> renders correctly', () => {
  const dispatchSpy = jest.fn();
  const wrapper = shallow(
    <HeaderComponent
      currentRoomName={currentRoomName}
      numberOfMembers={numberOfMembers}
      user={user}
      dispatch={dispatchSpy}
    />
  );
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
