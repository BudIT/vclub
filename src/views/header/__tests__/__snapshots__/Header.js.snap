exports[`test <Header /> renders correctly 1`] = `
<div
  className="header">
  <HeaderLeft
    currentRoomName="VIDEO"
    dispatch={[Function]}
    roomsNames={
      Array [
        "CHAT",
        "SHARING",
        "MEDIA",
        "WHITEBOARD",
      ]
    }
    user={
      Object {
        "id": "1",
        "master": true,
        "name": "User",
      }
    } />
  <HeaderRight
    dispatch={[Function]}
    numberOfMembers={5}
    showMemberPanel={false}
    user={
      Object {
        "id": "1",
        "master": true,
        "name": "User",
      }
    } />
</div>
`;
